File: client/src/entities/Task/model/types/Task.ts
import { Profile } from '@/entities/Profile';

export enum TaskStatus {
    CREATED = 'CREATED',
    PROGRESS = 'PROGRESS',
    REVIEW = 'REVIEW',
    COMPLETED = 'COMPLETED',
}

export enum TaskPriority {
    FEATURE = 'FEATURE',
    MEDIUM = 'MEDIUM',
    CRITICAL = 'CRITICAL',
}

export interface Task {
    id: string;
    title: string;
    description: string;
    status: TaskStatus;
    priority: TaskPriority;
    contributors: Profile[];
    createdAt: Date;
    updatedAt: Date;
    deadline: Date;
}

File: backend/server/internal/domain/types/models/task.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"server/internal/domain/types/enum"
	"time"
)

type Task struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Title       string             `json:"title" bson:"title"`
	Description string             `json:"description" bson:"description"`
	Status      enum.Status        `json:"status" bson:"status"`
	Priority    enum.Priority      `json:"priority" bson:"priority"`

	Contributors []primitive.ObjectID `bson:"contributors" json:"contributors"`

	Deadline time.Time `json:"deadline" bson:"deadline"`

	CreatedAt time.Time `json:"createdAt" bson:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt" bson:"updatedAt"`
}
